builtin/providers/aws/resource_aws_spot_fleet_request.go

```diff
@@ -564,8 +583,28 @@ func resourceAwsSpotFleetRequestCreate(d *schema.ResourceData, meta interface{})
 		DryRun:                 aws.Bool(false),
 	}
 
-	log.Printf("[DEBUG] Requesting spot fleet with these opts: %+v", spotFleetOpts)
-	resp, err := conn.RequestSpotFleet(spotFleetOpts)
+	fmt.Printf("[DEBUG] Requesting spot fleet with these opts: %+v", spotFleetOpts)
+
+	// Since IAM is eventually consistent, we retry creation as a newly created role may not
+	// take effect immediately, resulting in an InvalidSpotFleetRequestConfig error
+	var resp *ec2.RequestSpotFleetOutput
```

Comment:
This is actually a known issue with a lot of AWS related resource on Terraform. I can't just insert a sleep in my itests since terraform internally tries to parallelize the tasks with it's graph so as soon as it makes the AWS IAM role, it starts the Spot fleet request and there's no way to insert a delay between this step (the test is equivalent to running "terraform apply"). This seems to be the solution used by other resources (https://github.com/hashicorp/terraform/blob/bf8931b1aba05cbaa5639ddc9575a447a09341fc/builtin/providers/aws/resource_aws_lambda_permission.go#L102), so I ended up doing the same thing.